homeassistant/components/srp_energy/__init__.py

```diff
@@ -1,34 +1,48 @@
 """The SRP Energy integration."""
-import logging
-
 from srpenergy.client import SrpEnergyClient
 
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import CONF_ID, CONF_PASSWORD, CONF_USERNAME, Platform
+from homeassistant.const import (
+    CONF_ID,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    Platform,
+)
 from homeassistant.core import HomeAssistant
-from homeassistant.exceptions import ConfigEntryNotReady
-
-from .const import SRP_ENERGY_DOMAIN
-
-_LOGGER = logging.getLogger(__name__)
 
+from .const import (  # noqa: F401
+    ATTRIBUTION,
+    CONF_IS_TOU,
+    DEFAULT_NAME,
+    DOMAIN,
+    ICON,
+    LOGGER,
+    PHOENIX_TIME_ZONE,
+    SENSOR_NAME,
+    SENSOR_TYPE,
+)
 
 PLATFORMS = [Platform.SENSOR]
 
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up the SRP Energy component from a config entry."""
-    # Store an SrpEnergyClient object for your srp_energy to access
-    try:
-        srp_energy_client = SrpEnergyClient(
-            entry.data.get(CONF_ID),
-            entry.data.get(CONF_USERNAME),
-            entry.data.get(CONF_PASSWORD),
-        )
-        hass.data[SRP_ENERGY_DOMAIN] = srp_energy_client
-    except (Exception) as ex:
-        _LOGGER.error("Unable to connect to Srp Energy: %s", str(ex))
-        raise ConfigEntryNotReady from ex
+    api_account_id: str = entry.data[CONF_ID]
+    api_username: str = entry.data[CONF_USERNAME]
+    api_password: str = entry.data[CONF_PASSWORD]
+    name: str = entry.data[CONF_NAME]
+
+    LOGGER.debug("%s Using account_id %s", name, api_account_id)
+
+    api_instance = SrpEnergyClient(
```

Comment:
Why are we not checking if we can connect with the client anymore?

Reply:
This was a fix to the original code. Instantiating the client doesn't actually make any api calls. Only the client methods `srp_client.validate` and `srp_client.usage`

Reply:
We should move the coordinator setup to here and do a first refresh here to validate that credentials and connection works.

https://developers.home-assistant.io/docs/integration_fetching_data#coordinated-single-api-poll-for-data-for-all-entities

Reply:
That code is coming in the next iteration. This pr is just for multiple configs.